### ISSUE TITLE
 DCMTKJPEG_INCLUDE_DIR  is not well defined

### ISSUE THREAD

**Issue by weiliu620** [GeertLitjens/ASAP#2]
I spent some time to figure out how to set this variable in cmake. I found I can set it to: 

 DCMTKJPEG_INCLUDE_DIR            /home/weiliu/packages/dcmtk-3.6.1_20160216/build/include 

And that works. This is because many files in ASAP include something like "dcmtk/dcmjpeg/libijg8/jpeglib8.h", which assume I should set the above variable so the full path of the .h file is valid.

However, now the name of DCMTKJPEG_INCLUDE_DIR is confusing, because this variable should be the path of all include files of DCMTK, not just DCMJPEG. 

Also, I need to copy the libijg8 folder from dcmtk source tree to the installed folder, because the 'make install' command does not do that for me.  Not sure if this is a ASAP issue or a dcmtk issue.

**Comment by GeertLitjens** [comment]
Thank you for reporting, let me shed some light on why we need this library at all. The 8-bit jpeg library (ijg8) which is part of DCMTK is the only JPEG library I know of that has support for decoding lossless JPEG.

The include path is a bit tricky, because it differs if you build it yourself or if you use a system package (where often ijg8 is not even included). I will think about at least adding a more descriptive info-message to the parameter.

**Comment by weiliu620** [comment]
It would be good to add some documentation on how to build/configure dcmtk for this particular libijg8 thing, so other people will save time building it. 

Also, an unrelated question, is there any documentation/examples of how to use the ASAP library, particular the python wrapper for reading to/from numpy array? 

Thanks!

**Comment by hmeine** [comment]
libijg8 is also not included in the MacPorts dcmtk package (version 3.6.1_20160216_1).  The package offers several variants (optional parts), but none of them appear relevant to me here:
```
   cxx11: Compile with c++11 support
   debug: Enable debug binaries
   doc: Install documentation
[+]png: Enable png support
   private: Install private tags dictionary
   sound: Enable sndfile support
[+]ssl: Enable openssl support
[+]tiff: Enable tiff support
   universal: Build for multiple architectures
[+]xml: Enable xml support
```

**Comment by f-squirrel** [comment]
Did anyone find a way to compile/get dcmtk with libijg8 to MacOS?

**Comment by GeertLitjens** [comment]
You can disable the VSI support in CMake now, which removes the need for having libijg8, that might help.

**Comment by f-squirrel** [comment]
@GeertLitjens I did it but I get linkage error:

> [  7%] Built target core
[  8%] Linking CXX shared library libmultiresolutionimageinterface.dylib
Undefined symbols for architecture x86_64:
  "typeinfo for VSIImage", referenced from:
      MultiResolutionImageFactory::openImage(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in MultiResolutionImageFactory.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [io/multiresolutionimageinterface/libmultiresolutionimageinterface.dylib] Error 1
make[1]: *** [io/multiresolutionimageinterface/CMakeFiles/multiresolutionimageinterface.dir/all] Error 2
make: *** [all] Error 2

Thus I suppose VSI support is still needed somewhere.

Any ideas?

**Comment by GeertLitjens** [comment]
You are right, for some reason MSVC is not triggered by this. You can just remove the cast to VSIImage and include for the time being from that file. Please do note that in principle we do not support macOS at the moment, so we cannot help with Mac specific issues

**Comment by fkluknav** [comment]
Hi,
by looking at the sources of dcmtk, it seems that jpeglib8.h is not meant for client applications. It is in the sources and not in the headers for installation. Maybe using it is not the right thing to do? Maybe djijg8.h and djeijg8.h should be used instead?

**Comment by fkluknav** [comment]
A side note, using libraries available in common Linux distros would do wonders with usability of ASAP. A few days of tinkering with other people's computers to build it is often not possible. To package a new library for a linux distro is usually doable. To package a slightly modified version of a library that is already available in the distro is a lot of pain and mess.
I am not familiar with DCMTK, do not know yet if ijg8 can be really used through the usual public dcmjpeg interface of DCMTK, but it would be very, very nice.
